fedimint-bitcoind/src/lib.rs

```diff
@@ -45,7 +59,7 @@ pub fn register_bitcoind(kind: String, factory: DynBitcoindRpcFactory) {
     registry.insert(kind, factory);
 }
 
-pub trait IBitcoindRpcFactory: Debug {
+pub trait IBitcoindRpcFactory: Debug + Send + Sync {
```

Comment:
Unclear what `fn create` does and when it can fail. Does it immediately connect?

Interfaces are by their nature the most important to document well.

Reply:
Conceptually all the `*ModuleGen`s are also factories, might make sense to unify naming.